### PR TITLE
Development setup and documentation

### PR DESCRIPTION
**Pull request by AlDu2407**
- For development purposes a Docker compose file with a MySQL configuration has been added. It contains default configuration values that can be used with Urchin to create and start an development database.
- Documentation has been updated to ease up the setup process for developing Urchin.
- Added the missing MIT license to the repository.

### CODE REVIEW

**Comment by matheusgomes28** [comment]
Thanks Al! I'll take a look today, like the initial changes so far 😄

File: docker/mysqldb.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: '3.1'
+
+services:
+  urchindb:
+    image: mysql
+    container_name: urchindb
+    ports: 
+      - 3306:3306
+    env_file: 
+      - ../.dev.env
```
Comment (matheusgomes28) [inline docker/mysqldb.yml:10]: Let's add a newline here

File: .dev.env [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+MYSQL_PASSWORD=urchinpw
+
+# Urchin environment variables
+URCHIN_DATABASE_ADDRESS=localhost
+URCHIN_DATABASE_PORT=3306
+URCHIN_DATABASE_USER=urchin
+URCHIN_DATABASE_PASSWORD=urchinpw
+URCHIN_DATABASE_NAME=urchin
+URCHIN_WEBSERVER_PORT=8081
+URCHIN_IMAGE_DIRECTORY=./images
```
Comment (matheusgomes28) [inline .dev.env:14]: new line here

File: .gitignore [hunk 3]
```diff
@@ -2,3 +2,4 @@
 **/__debug*
 **/*_templ.go
 tmp/
+**/main
```
Comment (matheusgomes28) [inline .gitignore:5]: What's this ignore for?

**Review by matheusgomes28 — CHANGES_REQUESTED** [review]
Looks good Al, thanks for the changes! Some minor comments, mainly about the lack of docs for the toml config.

**Comment by AlDu2407** [comment]
Hi Mat, will implement your suggested changes later on and update the PR. Thanks for the feedback.

File: .gitignore (hunk 3 above)
Comment (AlDu2407) [reply .gitignore:5]: I think you can ignore this, I'll revert it. I think this is something that my IDE added when I ignored a compiled version of the application.

**Comment by AlDu2407** [comment]
Updated PR due to latest feedback and reverted specific parts of the PR.

File: urchin_config.toml [hunk 4]
```diff
@@ -16,3 +16,6 @@ database_name = "urchin"
 
 # port to run the webserver on
 webserver_port = 8080
+
+# Directory to use for storing uploaded images.
+image_dir = "./images"
```
Comment (matheusgomes28) [inline urchin_config.toml:21]: I guess just new line here to keep it consistent

**Review by matheusgomes28 — COMMENTED** [review]
Thank you, looks much better. I'll test tonight and assuming that the docker compose is all working, I'll merge it :)

File: docker/mysqldb.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+version: '3.1'
```
Comment (matheusgomes28) [inline docker/mysqldb.yml:1]: Hey Al, taking a second look at this PR, I realised that you only added the `mysql` container to this compose yml. If you take a look at the `docker-compose.yml` in the project repo, you can see that I start both the `urchin` app and the `mariadb` app.

Do you want to move some of that logic to this? Also, let's start a `mariadb` image as that's what we use for development.

Comment (AlDu2407) [reply docker/mysqldb.yml]: As mentioned, the `mariadb.yml` should be used only for local development. I moved the docker-compose file for full set-up into the `docker` directory and added all the files related to starting the application in a docker container within the same directory.

When developing you have now the option to either start the application in a full docker setup and work locally, or you can just spin up a development database and run the urchin from within VSCode with the given launch.json.

Comment (matheusgomes28) [reply docker/mysqldb.yml]: I like it, I agree with the changes. I didn't realise that it would be easier to spin up the container just with the Database.

**Review by matheusgomes28 — APPROVED** [review]
Thank you for the changes Al

**Pull request merged**